plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -128,6 +200,87 @@ public Page getNextPage()
         }
     }
 
+    @Override
+    public void deleteRows(Block rowIds)
+    {
+        RowBlock rows = resolveRowIdBlock(rowIds);
+        posDeleteSink.appendPage(new Page(rows.getPositionCount(), rows.getChildren().toArray(new Block[0])));
+    }
+
+    @Override
+    public void updateRows(Page page, List<Integer> columnValueAndRowIdChannels)
+    {
+        int batchSize = page.getPositionCount();
+        Block rowIdRawBlock = page.getBlock(columnValueAndRowIdChannels.get(columnValueAndRowIdChannels.size() - 1));
+        RowBlock rowIdBlock = resolveRowIdBlock(rowIdRawBlock);
+        Block[] rowIdFieldBlocks = rowIdBlock.getChildren().toArray(new Block[0]);
```

Comment:
`Block.getChildren()` is probably not correct. 
see https://github.com/trinodb/trino/pull/9354
you might want to check with @djsstarburst , but it seems ColumnarRow should be used here.